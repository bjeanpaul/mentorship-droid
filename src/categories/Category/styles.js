import { StyleSheet } from 'react-native';

import { FONT, FONT_WEIGHT } from 'src/constants/styles';
import colors from 'src/constants/colors';


export default StyleSheet.create({
  header: {
    height: 64,
  },
  back: {
    top: 20,
  },
  tabGroup: {
    height: 40,
    flexDirection: 'row',
  },
  tab: {
    flex: 1,
  },
  tabActive: {
    borderBottomWidth: 2,
    borderBottomColor: colors.CATEGORY_ACTIVE_TAB_BORDER,
  },
  tabTitle: {
    fontSize: 16,
    fontFamily: FONT.MEDIUM,
    fontWeight: FONT_WEIGHT.MEDIUM,
    color: colors.TEXT_LIGHT,
    opacity: 0.5,
  },
  tabTitleActive: {
    opacity: 1,
  },
});
