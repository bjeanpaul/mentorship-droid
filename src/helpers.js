import { find, isUndefined, partialRight as partial } from 'lodash';
import { getBaseURL } from 'src/configuration';
import { omitBy, isNull } from 'lodash';


export const omitNulls = d => omitBy(d, isNull);


export const trap = (type, fn) => v => v instanceof type
  ? Promise.resolve(v).then(fn)
  : Promise.reject(v);


export const switchError = cases => (e, ...xargs) => {
  const [type, fn] = find(cases, ([type, fn]) => e instanceof type) || [];
  return !isUndefined(fn)
    ? Promise.resolve().then(() => fn(e, ...xargs))
    : Promise.reject(e);
};


export const apiAction = ({
  method,
  request,
  success,
  failures,
}) => (...args) => (dispatch, { auth }) => Promise.resolve()
  .then(() => request(...args))
  .then(dispatch)
  .then(() => method(...args, auth))
  .then(partial(success, ...args), partial(switchError(failures), ...args))
  .then(dispatch);


export const staticAction = type => () => ({
  type
});


export const dataAction = type => data => ({
  type,
  payload: data
});


export const generateActionTypes = function generateActionNames(
  resource,
  operations = ['FETCH', 'CREATE', 'UPDATE', 'DELETE']
) {
  const actionTypes = {};
  const resourceUC = resource.toUpperCase();

  operations.forEach((operation) => {
    const operationUC = operation.toUpperCase();
    const request = `${resourceUC}_${operationUC}_REQUEST`;
    const success = `${resourceUC}_${operationUC}_SUCCESS`;
    const failure = `${resourceUC}_${operationUC}_FAILURE`;

    const operationLC = operation.toLowerCase();
    actionTypes[`${operationLC}Request`] = request;
    actionTypes[`${operationLC}Success`] = success;
    actionTypes[`${operationLC}Failure`] = failure;
  });
  return actionTypes;
};


export const filterActionTypes = function filterActionTypes(
  actionTypes,
  operation
) {
  const [request, success, failure] = [
    `${operation}Request`,
    `${operation}Success`,
    `${operation}Failure`,
  ];

  if (actionTypes[request] || actionTypes[success] || actionTypes[failure]) {
    return [
      actionTypes[request],
      actionTypes[success],
      actionTypes[failure],
    ];
  }
  return [];
};


// TODO: Delete
export const generateActionCreators = function generateActionCreators({
  resourcePath,
  actionTypes,
  requestOpts = {},
  normalizeJSON,
}) {
  const actionCreators = {};
  let url = `${getBaseURL()}/`;
  if (resourcePath) {
    url += `${resourcePath}/`;
  }

  const fetchActionTypes = filterActionTypes(actionTypes, 'fetch');
  if (fetchActionTypes.length === 3) {
    actionCreators.fetch = (onSuccess, onFailure) => ({
      type: '--generated fetch--',
      types: fetchActionTypes,
      url,
      requestOpts,
      normalizeJSON,
      onSuccess,
      onFailure,
    });
  }

  const createActionTypes = filterActionTypes(actionTypes, 'create');
  if (createActionTypes.length === 3) {
    actionCreators.create = ({
      body,
      onSuccess,
      onFailure,
    }) => ({
      type: '--generated create--',
      types: createActionTypes,
      url,
      requestOpts: { ...requestOpts,
        method: 'POST',
        body: JSON.stringify(body),
      },
      normalizeJSON,
      onSuccess,
      onFailure,
    });
  }

  // id is required as an argument to update a record;
  // we have a scenario with profiless, where it's
  // profile/{id}/image,
  // which is awkward...
  const updateActionTypes = filterActionTypes(actionTypes, 'update');
  if (updateActionTypes.length === 3) {
    actionCreators.update = ({
      id,
      body,
      onSuccess,
      onFailure,
    }) => ({
      type: '--generated update--',
      types: updateActionTypes,
      url: `${url}${id}/`,
      requestOpts: { ...requestOpts,
        method: 'PUT',
        body: JSON.stringify(body),
      },
      normalizeJSON,
      onSuccess,
      onFailure,
    });
  }


  return actionCreators;
};
