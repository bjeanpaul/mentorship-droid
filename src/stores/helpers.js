import moment from 'moment';
import { values, isUndefined, sortBy, filter, first } from 'lodash';


export const getAuthUserProfile = ({
  auth: { profileId },
  entities: { profiles },
}) => !isUndefined(profileId)
  ? profiles[profileId]
  : null;


export const getContext = state => {
  const { auth: { auth } } = state;

  return {
    auth,
    profile: getAuthUserProfile(state),
  };
};


// TODO sort?
export const getCategories = ({ entities: { categories } }) => values(categories);


// TODO sort?
export const getCategoryActivities = ({
  entities: { activities },
}, targetCategoryId) => (
  values(activities)
    .filter(({ category }) => category === targetCategoryId));


export const getCategory = ({ entities: { categories } }, id) => categories[id];


export const getActivity = ({ entities: { activities } }, id) => activities[id];


export const getScheduledCalls = ({ entities: { scheduledCalls } }) => values(scheduledCalls);


export const getScheduledCall = ({ entities: { scheduledCalls } }, id) => scheduledCalls[id];


export const getScheduledCallActivity = (state, id) => {
  const scheduledCall = getScheduledCall(state, id);
  return scheduledCall && scheduledCall.activity && getActivity(state, scheduledCall.activity);
};


export const getEvents = ({ entities: { events } }) => values(events);


export const getActivityCallNotes = ({
  entities: { callNotes },
}, activityId) => (
  filter(callNotes, { callActivity: activityId }));


export const getCall = ({ entities: { calls } }, id) => calls[id];


export const getCallNote = ({ entities: { callNotes } }, id) => callNotes[id];


export const getNextScheduledCall = (state, predicate = {}, time = Date.now()) => {
  let res = sortBy(getScheduledCalls(state), ({ callTime }) => callTime);
  res = res.filter(({ callTime }) => moment(callTime).isSameOrAfter(time));
  res = filter(res, predicate);
  return first(res);
};
