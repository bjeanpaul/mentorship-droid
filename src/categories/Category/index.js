import React, { PropTypes } from 'react';
import { BaseView, Header, HeaderIcon, Title } from 'src/components';
import styles from './styles';


const Category = ({
  category: {
    title,
    color,
  },
}) => (
  <BaseView>
    <Header style={[styles.header, { backgroundColor: color }]}>
      <HeaderIcon type={HeaderIcon.types.backLight} style={styles.back} />
      <Title theme={Title.themes.light}>{title}</Title>
    </Header>
  </BaseView>
);


Category.propTypes = {
  category: PropTypes.object.isRequired,
};


export default Category;
