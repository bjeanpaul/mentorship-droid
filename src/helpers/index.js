import { getBaseURL } from 'src/configuration';
import { omitBy, isNull } from 'lodash';


export const conj = (...args) => Object.assign({}, ...args);


export const omitNulls = d => omitBy(d, isNull);


export const generateActionTypes = function generateActionNames(
  resource,
  operations = ['FETCH', 'CREATE', 'UPDATE', 'DELETE']
) {
  const actionTypes = {};
  const resourceUC = resource.toUpperCase();

  operations.forEach((operation) => {
    const operationUC = operation.toUpperCase();
    const request = `${resourceUC}_${operationUC}_REQUEST`;
    const success = `${resourceUC}_${operationUC}_SUCCESS`;
    const failure = `${resourceUC}_${operationUC}_FAILURE`;

    const operationLC = operation.toLowerCase();
    actionTypes[`${operationLC}Request`] = request;
    actionTypes[`${operationLC}Success`] = success;
    actionTypes[`${operationLC}Failure`] = failure;
  });
  return actionTypes;
};


export const filterActionTypes = function filterActionTypes(
  actionTypes,
  operation
) {
  const [request, success, failure] = [
    `${operation}Request`,
    `${operation}Success`,
    `${operation}Failure`,
  ];

  if (actionTypes[request] || actionTypes[success] || actionTypes[failure]) {
    return [
      actionTypes[request],
      actionTypes[success],
      actionTypes[failure],
    ];
  }
  return [];
};


// TODO: Delete
export const generateActionCreators = function generateActionCreators({
  resourcePath,
  actionTypes,
  requestOpts = {},
  normalizeJSON,
}) {
  const actionCreators = {};
  let url = `${getBaseURL()}/`;
  if (resourcePath) {
    url += `${resourcePath}/`;
  }

  const fetchActionTypes = filterActionTypes(actionTypes, 'fetch');
  if (fetchActionTypes.length === 3) {
    actionCreators.fetch = (onSuccess, onFailure) => ({
      type: '--generated fetch--',
      types: fetchActionTypes,
      url,
      requestOpts,
      normalizeJSON,
      onSuccess,
      onFailure,
    });
  }

  const createActionTypes = filterActionTypes(actionTypes, 'create');
  if (createActionTypes.length === 3) {
    actionCreators.create = ({
      body,
      onSuccess,
      onFailure,
    }) => ({
      type: '--generated create--',
      types: createActionTypes,
      url,
      requestOpts: { ...requestOpts,
        method: 'POST',
        body: JSON.stringify(body),
      },
      normalizeJSON,
      onSuccess,
      onFailure,
    });
  }

  // id is required as an argument to update a record;
  // we have a scenario with profiless, where it's
  // profile/{id}/image,
  // which is awkward...
  const updateActionTypes = filterActionTypes(actionTypes, 'update');
  if (updateActionTypes.length === 3) {
    actionCreators.update = ({
      id,
      body,
      onSuccess,
      onFailure,
    }) => ({
      type: '--generated update--',
      types: updateActionTypes,
      url: `${url}${id}/`,
      requestOpts: { ...requestOpts,
        method: 'PUT',
        body: JSON.stringify(body),
      },
      normalizeJSON,
      onSuccess,
      onFailure,
    });
  }


  return actionCreators;
};
