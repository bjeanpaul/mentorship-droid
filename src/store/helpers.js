import moment from 'moment';
import { values, isUndefined, sortBy, filter, first, orderBy } from 'lodash';

import { getCurrent } from 'src/navigationHelpers';
import { TYPES_TO_COLLECTIONS } from 'src/constants/events';


export const getAuthUserProfile = ({
  auth: { profileId },
  entities: { profiles },
}) => !isUndefined(profileId)
  ? profiles[profileId]
  : null;


export const getContext = state => {
  const { auth: { auth } } = state;

  return {
    auth,
    profile: getAuthUserProfile(state),
  };
};


export const getCategories = ({ entities }) =>
  sortBy(values(entities.categories), 'ordinal');


export const getCategoryActivities = ({ entities }, targetCategoryId) => {
  const activities = values(entities.activities)
    .filter(({ category }) => category === targetCategoryId);

  return sortBy(activities, 'ordinal');
};


export const getCategory = ({ entities: { categories } }, id) =>
  categories[id];


export const getActivity = ({ entities: { activities } }, id) =>
  activities[id];


export const getScheduledCalls = ({ entities: { scheduledCalls } }) =>
  values(scheduledCalls);


export const getScheduledCall = ({ entities: { scheduledCalls } }, id) =>
  scheduledCalls[id];


export const getScheduledCallActivity = (state, id) => {
  const scheduledCall = getScheduledCall(state, id);
  return scheduledCall
      && scheduledCall.activity
      && getActivity(state, scheduledCall.activity);
};


export const getActivityCallNotes = ({
  entities: { callNotes },
}, activityId) =>
  values(callNotes)
    .filter(({ callActivityId }) => callActivityId === activityId);


export const getCallNotes = ({
  entities: { callNotes },
}) =>
  sortBy(callNotes, ({ callStartTime }) => +moment(callStartTime));


export const getCall = ({ entities: { calls } }, id) => calls[id];


export const getCallNote = ({ entities: { callNotes } }, id) => callNotes[id];


export const getNextScheduledCall = (state, predicate = {}, time = Date.now()) => {
  let res = sortBy(getScheduledCalls(state), ({ callTime }) => +moment(callTime));
  res = res.filter(({ callTime }) => moment(callTime).isSameOrAfter(time));
  res = filter(res, predicate);
  return first(res);
};


export const getScheduledCallsBetween = (state, start, end, time = Date.now()) => {
  // for some reason, moment durations are behaving inconsistently in tests,
  // thus the lack of moment usage here
  const startTime = +moment(time) - start;
  const endTime = +moment(time) + end;

  return getScheduledCalls(state)
    .filter(({ callTime }) => {
      const t = +moment(callTime);
      return startTime <= t && t <= endTime;
    });
};


const getEventObject = (state, event) => {
  const collection = state.entities[TYPES_TO_COLLECTIONS[event.eventType]];
  return collection && collection[event.objectId];
};


// TODO return the associated entitites to make the containers' jobs easier
export const getEvents = (state) =>
  values(state.entities.events)
    .filter(event => getEventObject(state, event));


export const getMessages = ({ entities }) => []
  .concat(values(entities.messages))
  .concat(values(entities.pendingMessages));


export const getMessage = ({ entities }, id) => entities.messages[id];


export const getActiveTopRoute = store => getCurrent(store.navigation.top);


export const getBlogPost = ({ entities }, id) => entities.blogPosts[id];


export const getBlogPosts = ({ entities }) =>
  orderBy(entities.blogPosts, [({ createdAt }) => +moment(createdAt)], ['desc']);
