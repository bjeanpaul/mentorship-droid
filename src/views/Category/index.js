import React, { Component, PropTypes } from 'react';
import { View, TouchableNativeFeedback } from 'react-native';

import { BaseView, Header, HeaderIcon, Text } from 'src/components';
import ActivityList from 'src/views/ActivityList';
import CategoryAbout from 'src/views/CategoryAbout';
import styles from './styles';


const TAB_ACTIVITIES = 'TAB_ACTIVITIES';
const TAB_ABOUT = 'TAB_ABOUT';


class Category extends Component {
  constructor(props) {
    super(props);

    const {
      initialActiveTab = TAB_ABOUT,
    } = props;

    this.state = {
      activeTab: initialActiveTab,
    };
  }

  onTabChange(tab) {
    this.setState({ activeTab: tab });
  }

  render() {
    const {
      category,
      ...props,
    } = this.props;

    const {
      color,
      title,
    } = category;

    return (
      <BaseView>
        <Header style={[styles.header, { backgroundColor: color }]}>
          <HeaderIcon type={HeaderIcon.types.backLight} style={styles.back} />
          <Text style={[Text.types.title, Text.themes.light]}>{title}</Text>
        </Header>

        <View style={[styles.tabGroup, { backgroundColor: color }]}>
          <CategoryTab
            tabId={TAB_ACTIVITIES}
            title="Activities"
            onPress={() => this.onTabChange(TAB_ACTIVITIES)}
            active={this.state.activeTab === TAB_ACTIVITIES}
          />

          <CategoryTab
            tabId={TAB_ABOUT}
            title="About"
            onPress={() => this.onTabChange(TAB_ABOUT)}
            active={this.state.activeTab === TAB_ABOUT}
          />
        </View>

        <CategoryTabBody
          tab={this.state.activeTab}
          category={category}
          {...props}
        />
      </BaseView>
    );
  }
}


const CategoryTab = ({
  tabId,
  active,
  title,
  onPress,
}) => (
  <TouchableNativeFeedback onPress={onPress} tabId={tabId}>
    <View style={[styles.tab, active && styles.tabIsActive]}>
      <Text style={[styles.tabTitle, active && styles.tabTitleIsActive]}>{title}</Text>
    </View>
  </TouchableNativeFeedback>
);


const CategoryTabBody = ({
  tab,
  ...props,
}) => {
  switch (tab) {
    case TAB_ACTIVITIES:
      return <ActivityList {...props} />;

    case TAB_ABOUT:
      return <CategoryAbout {...props} />;

    default:
      return null;
  }
};


const tabPropType = PropTypes.oneOf([TAB_ACTIVITIES, TAB_ABOUT]);


Category.propTypes = {
  category: PropTypes.object.isRequired,
  initialActiveTab: tabPropType,
};


CategoryTab.propTypes = {
  tabId: PropTypes.string.isRequired,
  active: PropTypes.bool.isRequired,
  title: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
};


CategoryTabBody.propTypes = {
  tab: tabPropType,
};


export default Category;

export {
  TAB_ACTIVITIES,
  TAB_ABOUT,
};
