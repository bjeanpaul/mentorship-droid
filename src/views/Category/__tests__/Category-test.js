import { noop } from 'lodash';
import React from 'react';
import Category, { TAB_ACTIVITIES, TAB_ABOUT } from 'src/views/Category';
import { uidEquals, fakeCategory, fakeActivity } from 'app/scripts/helpers';


describe('Category', () => {
  function createComponent(props = {}) {
    return (
      <Category
        category={fakeCategory({
          title: 'Level',
          color: '#97c13c',
        })}
        activities={[fakeActivity()]}
        onActivityPress={noop}
        onBackPress={noop}
        {...props}
      />
    );
  }

  it('should render', () => {
    const el = render(createComponent());
    expect(el).toMatchSnapshot();
  });

  it('should render the currently active tab', () => {
    let el;
    el = render(createComponent({ initialPropType: TAB_ABOUT }));
    expect(el).toMatchSnapshot();

    el = render(createComponent({ initialPropType: TAB_ACTIVITIES }));
    expect(el).toMatchSnapshot();
  });

  it('should call onBackPress when the back button is pressed', () => {
    const onBackPress = jest.fn();
    const el = shallow(createComponent({ onBackPress }));

    el.findWhere(uidEquals('back'))
      .simulate('press');

    expect(onBackPress.mock.calls).toEqual([[]]);
  });
  it('should call onTabChange when a tab is pressed', () => {
    const el = shallow(createComponent());
    const onTabChange = jest.fn();
    el.instance().onTabChange = onTabChange;

    el.findWhere(node => node.prop('tabId') === TAB_ABOUT)
      .simulate('press');

    expect(onTabChange.mock.calls).toEqual([
      [TAB_ABOUT],
    ]);

    el.findWhere(node => node.prop('tabId') === TAB_ACTIVITIES)
      .simulate('press');

    expect(onTabChange.mock.calls).toEqual([
      [TAB_ABOUT],
      [TAB_ACTIVITIES],
    ]);
  });
});
