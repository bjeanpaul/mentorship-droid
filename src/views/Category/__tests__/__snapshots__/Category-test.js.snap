exports[`Category should render 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
      },
      undefined,
    ]
  }>
  <View
    style={
      Array [
        Object {
          "height": 72,
          "justifyContent": "center",
          "paddingLeft": 16,
          "paddingRight": 16,
        },
        Array [
          Object {
            "height": 64,
          },
          Object {
            "backgroundColor": "#97c13c",
          },
        ],
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#003035",
            "fontFamily": "Brandon Text",
          },
          Object {
            "fontSize": 13,
            "textAlign": "center",
          },
          Array [
            Object {
              "fontFamily": "Brandon Text_medium",
              "fontSize": 20,
              "fontWeight": "400",
            },
            Object {
              "color": "#fff",
            },
          ],
        ]
      }>
      Level
    </Text>
    <TouchableNativeFeedback
      onPress={[Function]}>
      <View
        style={
          Array [
            Object {
              "position": "absolute",
              "top": 24,
            },
            Object {
              "top": 20,
            },
          ]
        }>
        <Image
          source={1}
          style={
            Array [
              Object {
                "height": 24,
                "width": 24,
              },
              Object {},
            ]
          } />
      </View>
    </TouchableNativeFeedback>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "height": 40,
        },
        Object {
          "backgroundColor": "#97c13c",
        },
      ]
    }>
    <TouchableNativeFeedback
      onPress={[Function]}
      tabId="TAB_ACTIVITIES">
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "borderBottomColor": "rgba(0, 0, 0, 0.3)",
              "borderBottomWidth": 2,
            },
          ]
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#003035",
                "fontFamily": "Brandon Text",
              },
              Object {
                "fontSize": 13,
                "textAlign": "center",
              },
              Array [
                Object {
                  "color": "#fff",
                  "fontFamily": "Brandon Text_medium",
                  "fontSize": 16,
                  "fontWeight": "400",
                  "opacity": 0.5,
                },
                Object {
                  "opacity": 1,
                },
              ],
            ]
          }>
          Activities
        </Text>
      </View>
    </TouchableNativeFeedback>
    <TouchableNativeFeedback
      onPress={[Function]}
      tabId="TAB_ABOUT">
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            false,
          ]
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#003035",
                "fontFamily": "Brandon Text",
              },
              Object {
                "fontSize": 13,
                "textAlign": "center",
              },
              Array [
                Object {
                  "color": "#fff",
                  "fontFamily": "Brandon Text_medium",
                  "fontSize": 16,
                  "fontWeight": "400",
                  "opacity": 0.5,
                },
                false,
              ],
            ]
          }>
          About
        </Text>
      </View>
    </TouchableNativeFeedback>
  </View>
  <ScrollView>
    <TouchableNativeFeedback
      onPress={[Function]}
      uid="activity:1">
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "borderBottomColor": "#dfe5e6",
            "borderBottomWidth": 1,
            "height": 120,
            "padding": 16,
          }
        }>
        <View
          style={
            Object {
              "position": "absolute",
              "top": 24,
            }
          }>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 72,
                  "height": 72,
                  "justifyContent": "center",
                  "width": 72,
                },
                Object {
                  "backgroundColor": "#ebebeb",
                },
              ]
            }>
            <Image
              source={1}
              style={
                Object {
                  "borderRadius": 72,
                  "height": 72,
                  "width": 72,
                }
              } />
          </View>
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={3}
          style={
            Array [
              Object {
                "color": "#003035",
                "fontFamily": "Brandon Text",
              },
              Object {
                "fontSize": 13,
                "textAlign": "center",
              },
              Object {
                "fontSize": 18,
                "marginLeft": 104,
                "textAlign": "left",
              },
            ]
          }>
          Activity 3
        </Text>
      </View>
    </TouchableNativeFeedback>
  </ScrollView>
</View>
`;

exports[`Category should render the currently active tab 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
      },
      undefined,
    ]
  }>
  <View
    style={
      Array [
        Object {
          "height": 72,
          "justifyContent": "center",
          "paddingLeft": 16,
          "paddingRight": 16,
        },
        Array [
          Object {
            "height": 64,
          },
          Object {
            "backgroundColor": "#97c13c",
          },
        ],
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#003035",
            "fontFamily": "Brandon Text",
          },
          Object {
            "fontSize": 13,
            "textAlign": "center",
          },
          Array [
            Object {
              "fontFamily": "Brandon Text_medium",
              "fontSize": 20,
              "fontWeight": "400",
            },
            Object {
              "color": "#fff",
            },
          ],
        ]
      }>
      Level
    </Text>
    <TouchableNativeFeedback
      onPress={[Function]}>
      <View
        style={
          Array [
            Object {
              "position": "absolute",
              "top": 24,
            },
            Object {
              "top": 20,
            },
          ]
        }>
        <Image
          source={1}
          style={
            Array [
              Object {
                "height": 24,
                "width": 24,
              },
              Object {},
            ]
          } />
      </View>
    </TouchableNativeFeedback>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "height": 40,
        },
        Object {
          "backgroundColor": "#97c13c",
        },
      ]
    }>
    <TouchableNativeFeedback
      onPress={[Function]}
      tabId="TAB_ACTIVITIES">
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "borderBottomColor": "rgba(0, 0, 0, 0.3)",
              "borderBottomWidth": 2,
            },
          ]
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#003035",
                "fontFamily": "Brandon Text",
              },
              Object {
                "fontSize": 13,
                "textAlign": "center",
              },
              Array [
                Object {
                  "color": "#fff",
                  "fontFamily": "Brandon Text_medium",
                  "fontSize": 16,
                  "fontWeight": "400",
                  "opacity": 0.5,
                },
                Object {
                  "opacity": 1,
                },
              ],
            ]
          }>
          Activities
        </Text>
      </View>
    </TouchableNativeFeedback>
    <TouchableNativeFeedback
      onPress={[Function]}
      tabId="TAB_ABOUT">
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            false,
          ]
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#003035",
                "fontFamily": "Brandon Text",
              },
              Object {
                "fontSize": 13,
                "textAlign": "center",
              },
              Array [
                Object {
                  "color": "#fff",
                  "fontFamily": "Brandon Text_medium",
                  "fontSize": 16,
                  "fontWeight": "400",
                  "opacity": 0.5,
                },
                false,
              ],
            ]
          }>
          About
        </Text>
      </View>
    </TouchableNativeFeedback>
  </View>
  <ScrollView>
    <TouchableNativeFeedback
      onPress={[Function]}
      uid="activity:1">
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "borderBottomColor": "#dfe5e6",
            "borderBottomWidth": 1,
            "height": 120,
            "padding": 16,
          }
        }>
        <View
          style={
            Object {
              "position": "absolute",
              "top": 24,
            }
          }>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 72,
                  "height": 72,
                  "justifyContent": "center",
                  "width": 72,
                },
                Object {
                  "backgroundColor": "#ebebeb",
                },
              ]
            }>
            <Image
              source={1}
              style={
                Object {
                  "borderRadius": 72,
                  "height": 72,
                  "width": 72,
                }
              } />
          </View>
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={3}
          style={
            Array [
              Object {
                "color": "#003035",
                "fontFamily": "Brandon Text",
              },
              Object {
                "fontSize": 13,
                "textAlign": "center",
              },
              Object {
                "fontSize": 18,
                "marginLeft": 104,
                "textAlign": "left",
              },
            ]
          }>
          Activity 3
        </Text>
      </View>
    </TouchableNativeFeedback>
  </ScrollView>
</View>
`;

exports[`Category should render the currently active tab 2`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
      },
      undefined,
    ]
  }>
  <View
    style={
      Array [
        Object {
          "height": 72,
          "justifyContent": "center",
          "paddingLeft": 16,
          "paddingRight": 16,
        },
        Array [
          Object {
            "height": 64,
          },
          Object {
            "backgroundColor": "#97c13c",
          },
        ],
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#003035",
            "fontFamily": "Brandon Text",
          },
          Object {
            "fontSize": 13,
            "textAlign": "center",
          },
          Array [
            Object {
              "fontFamily": "Brandon Text_medium",
              "fontSize": 20,
              "fontWeight": "400",
            },
            Object {
              "color": "#fff",
            },
          ],
        ]
      }>
      Level
    </Text>
    <TouchableNativeFeedback
      onPress={[Function]}>
      <View
        style={
          Array [
            Object {
              "position": "absolute",
              "top": 24,
            },
            Object {
              "top": 20,
            },
          ]
        }>
        <Image
          source={1}
          style={
            Array [
              Object {
                "height": 24,
                "width": 24,
              },
              Object {},
            ]
          } />
      </View>
    </TouchableNativeFeedback>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "height": 40,
        },
        Object {
          "backgroundColor": "#97c13c",
        },
      ]
    }>
    <TouchableNativeFeedback
      onPress={[Function]}
      tabId="TAB_ACTIVITIES">
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "borderBottomColor": "rgba(0, 0, 0, 0.3)",
              "borderBottomWidth": 2,
            },
          ]
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#003035",
                "fontFamily": "Brandon Text",
              },
              Object {
                "fontSize": 13,
                "textAlign": "center",
              },
              Array [
                Object {
                  "color": "#fff",
                  "fontFamily": "Brandon Text_medium",
                  "fontSize": 16,
                  "fontWeight": "400",
                  "opacity": 0.5,
                },
                Object {
                  "opacity": 1,
                },
              ],
            ]
          }>
          Activities
        </Text>
      </View>
    </TouchableNativeFeedback>
    <TouchableNativeFeedback
      onPress={[Function]}
      tabId="TAB_ABOUT">
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            false,
          ]
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#003035",
                "fontFamily": "Brandon Text",
              },
              Object {
                "fontSize": 13,
                "textAlign": "center",
              },
              Array [
                Object {
                  "color": "#fff",
                  "fontFamily": "Brandon Text_medium",
                  "fontSize": 16,
                  "fontWeight": "400",
                  "opacity": 0.5,
                },
                false,
              ],
            ]
          }>
          About
        </Text>
      </View>
    </TouchableNativeFeedback>
  </View>
  <ScrollView>
    <TouchableNativeFeedback
      onPress={[Function]}
      uid="activity:1">
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "borderBottomColor": "#dfe5e6",
            "borderBottomWidth": 1,
            "height": 120,
            "padding": 16,
          }
        }>
        <View
          style={
            Object {
              "position": "absolute",
              "top": 24,
            }
          }>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 72,
                  "height": 72,
                  "justifyContent": "center",
                  "width": 72,
                },
                Object {
                  "backgroundColor": "#ebebeb",
                },
              ]
            }>
            <Image
              source={1}
              style={
                Object {
                  "borderRadius": 72,
                  "height": 72,
                  "width": 72,
                }
              } />
          </View>
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={3}
          style={
            Array [
              Object {
                "color": "#003035",
                "fontFamily": "Brandon Text",
              },
              Object {
                "fontSize": 13,
                "textAlign": "center",
              },
              Object {
                "fontSize": 18,
                "marginLeft": 104,
                "textAlign": "left",
              },
            ]
          }>
          Activity 3
        </Text>
      </View>
    </TouchableNativeFeedback>
  </ScrollView>
</View>
`;
